src/chapters/unit-testing/tdd.rst

```diff
@@ -0,0 +1,197 @@
+referenced code that does not exist yet.
+
+.. figure:: figures/red-green-tests.png
+      :alt: Graphic of testing output with red 'x' indicating we failed our NumberOfWheels test.
+
+      Failed the ``NumberOfWheels`` test.  Let's refactor our code.
+
+
+Finally, write code to pass the new test. In the earlier chapters, this is
+where you started, but with TDD writing new code is the *last* step.
+
+To make the new test pass, a property must be created that will implement a ``NumberOfWheels`` property.
+
+
+.. admonition:: Example
+   
+   .. sourcecode:: csharp
+      :lineno-start: 10 
+
+      //in the Car class
+
+      public int NumberOfWheels { get; set; }
+
+Now that we have updated the ``Car`` properties, we should look at our constructor.
+Add ``NumberOfWheels`` to the constructor.  
+Once this is complete, check your ``Car`` objects.  
+You will most likely see some red squiggles under your ``new Car`` objects.  
+Update your ``Car`` objects accordingly. 
+
+.. admonition:: Example
+
+   .. sourcecode:: csharp
+      :lineno-start: 20
+
+      //within the CarTests class
+
+      [TestMethod]
+      public void TestNumberOfWheels()
+      {
+         Car test_car = new Car("Toyota", "Prius", 10, 50, 4);
+         Assert.AreEqual(4, test_car.NumberOfWheels);
+      }
+
+Now when you run your tests, they should all pass.  Great job!
+
+.. figure:: figures/all-green-tests.png
+      :alt: Screenshot of the testing output screen showing all green checks.
+
+      All green!  All tests passed!
+
+
+Coding this way builds confidence in your work. No matter how large your code
+base may get, you know that each part has a test to validate its functionality.
+
+.. admonition:: Example
+
+   Now that we have one passing test for our data parser project, we could
+   confidently move on to writing tests and code for any other remaining features.
+
+Red, Green, Refactor
+--------------------
+
+.. index:: ! red green refactor
+
+.. index::
+   single: TDD; red green refactor
+
+.. index::
+   single: TDD; red, green, refactor
+
+.. index::
+   single: unit testing; red green refactor
+
+While adding new features and making our code work is the main goal, we also
+want to write readable, efficient code that makes us proud. The **red, green,
+refactor** mantra describes the process of writing tests, seeing them pass, and
+then making the code better. As the name suggests, the cycle consists of three
+steps. Red refers to test results that fail, while green represents tests that
+pass. The colors refer to test results which are often styled with red for
+failing tests and green for passing tests.
+
+#. Red -> Write a failing test.
+#. Green -> Make it pass by implementing the code.
+#. Refactor -> Make the code better.
+
+
+   .. figure:: figures/red-green-refactor.png
+      :alt: Graphic showing the cycle of phases from red the writing test, green making the test pass, and blue of refactoring code to be better which points back to red.
+
```

Comment:
I find this section confusing, this link helped me : https://mbauza.medium.com/red-green-refactor-1a3fb160e649

specifically lines 170 -172

Reply:
i added more intro into this can called it a methodology.  The specific lines you reference now contain links to the previous chapter to demo that they already did this once.